### PR TITLE
Removed check for iceGatheringState to be complete for bypassing gathering ice candidate again

### PR DESCRIPTION
**Pull request by anuranduttaroy**
**Issue #:** 
- https://github.com/aws/amazon-chime-sdk-js/issues/609
- https://github.com/aws/amazon-chime-sdk-js/issues/705

**Description of changes:**
In Firefox, the process of ICE Candidate Gathering was taking a long time to go to the `complete` state. In the JS SDK, there was a check to see if previous ICE Candidate Gathering request was complete or not in order to make a decision whether to bypass the new ICE Candidate Gathering request. We decided to remove this check as it was not making any difference since the SDK was not having any actions if any ICE Candidate arrived at a later point in time. It's all handled by the browser. 

Old: 
```
    if (
      this.context.peer.iceGatheringState === 'complete' &&
      new DefaultSDP(this.context.peer.localDescription.sdp).hasCandidates()
    )
```

New: 
```
if (new DefaultSDP(this.context.peer.localDescription.sdp).hasCandidates())
```

This delay in iceGatheringState to be marked as complete in Firefox was resulting in the delay in showing up the sender's video to remote participants.

**Result:**
Old:
```
FinishGatheringICECandidatesTask took 201 ms
FinishGatheringICECandidatesTask took 10045 ms
FinishGatheringICECandidatesTask took 2 ms
```

New:
```
FinishGatheringICECandidatesTask took 206 ms 
FinishGatheringICECandidatesTask took 0 ms 
FinishGatheringICECandidatesTask took 1 ms
```

**Testing**

1. Have you successfully run `npm run build:release` locally?
Yes.

2. How did you test these changes?
Tested using the demo app (locally & deployed the serverless app) to see any issues. (Specially in Safari). Did not see any issues.

3. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
No. Internal working is modified.

4. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
No. This is a change in how SDK works internally and have verified it with multiple members of the Video team.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by XHatan — DISMISSED** [review]

**Comment by XHatan** [comment]
Could you ask https://github.com/Sean-Der to take a look as well  ?

**Review by XHatan — DISMISSED** [review]

**Comment by devalevenkatesh** [comment]
Closing until release.

**Review by XHatan — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,11 +5,22 @@ All notable changes to this project will be documented in this file.
...
+### Removed
+
+### Fixed
+
 ## [1.20.2] - 2020-10-20
 ### Added
 
 ### Changed
 - [Documentation] Update README to add information about `tileState.active`
+- Removed check for iceGatheringState to be complete for bypassing gathering ice candidate again
```
Comment (simmkyu) [inline CHANGELOG.md:23]: Remove this line from the previous changelog

**Review by ltrung — APPROVED** [review]

File: src/task/FinishGatheringICECandidatesTask.ts [hunk 2]
```diff
@@ -76,9 +76,13 @@ export default class FinishGatheringICECandidatesTask extends BaseTask {
       );
     }
 
-    // To bypass waiting for events, it is required that "icegatheringstate" to be complete and sdp to have candidate
+    /*
+     * To bypass waiting for events, it is required that "icegatheringstate" to be complete and sdp to have candidate
+     * For Firefox, it takes long for iceGatheringState === 'complete'
```
Comment (simmkyu) [inline src/task/FinishGatheringICECandidatesTask.ts:81]: I think we need a bit more explanation, e.g. why only Firefox but not for other supported browsers?

**Review by tenge-amzn — APPROVED** [review]

**Review by simmkyu — APPROVED** [review]

**Pull request merged**